cassandra-bridge/src/main/java/org/apache/cassandra/spark/data/DataLayer.java

```diff
@@ -164,6 +164,11 @@ public CassandraVersion version()
 
     public abstract boolean isInPartition(int partitionId, BigInteger token, ByteBuffer key);
 
+    /**
+     * @return a TimeProvider
+     */
+    public abstract TimeProvider timeProvider();
```

Comment:
Do not return DEFAULT time prover. The concrete implementation should return the correct value.